pkg/clusterversion/clusterversion_test.go

```diff
@@ -1 +1,353 @@
 package clusterversion
+
+import (
+	"testing"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	corev1 "github.com/openshift/api/config/v1"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Common variables.
+var (
+	testSettings          *clients.Settings
+	updateChannel         = "test-channel"
+	updateVersion         = "4.9.0"
+	updateImage           = "4.9.0-image"
+	stream                = "Z"
+	availableCondition    = "Available"
+	desiredVersion        = "4.8.0"
+	defaultClusterVersion = "4.8.0"
+	defaultImage          = "4.8.0-image"
+)
+
+// Helper function to create a Builder with a clientfunc.
+func NewBuilder(apiClient *clients.Settings, name string) *Builder {
+	glog.V(100).Infof(
+		"Initializing new clusterversion structure with the following params: %s",
+		name)
+
+	if apiClient == nil {
+		glog.V(100).Infof("clusterversion 'apiClient' cannot be empty")
+
+		return nil
+	}
+
+	builder := Builder{
+		apiClient: apiClient,
+		Definition: &corev1.ClusterVersion{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: defaultClusterVersion,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the clusterversion is empty")
+
+		builder.errorMsg = "clusterversion 'name' cannot be empty"
+	}
+
+	return &builder
+}
+
+func TestClusterVersionPull(t *testing.T) {
+	testCases := []struct {
+		clusterVersionName  string
+		expectedError       bool
+		addToRuntimeObjects bool
+		client              bool
+		expectedErrorText   string
+	}{
+		{
+			clusterVersionName:  defaultClusterVersion,
+			expectedError:       false,
+			addToRuntimeObjects: true,
+			client:              true,
+			expectedErrorText:   "",
+		},
+		{
+			clusterVersionName:  "",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			client:              true,
+			expectedErrorText:   "clusterversion name cannot be empty",
```

Comment:
Something wrong with the test case. The error has to be 
`"clusterversion object version does not exist"`